core/src/main/java/org/apache/brooklyn/core/sensor/ReleaseableLatch.java

```diff
@@ -75,7 +89,11 @@ public void acquire(Entity caller) {
 
             @Override
             public void release(Entity caller) {
-                sem.release();
+                try {
+                    sem.release();
+                } finally {
+                    ownerEntities.remove(caller);
```

Comment:
I think this is fine. If we wanted to get more complicated, we could use a MultiMap or some such so that we know how many times the entity has called acquire(). But that doesn't feel worth the complexity, so I like the code as-is.